### PR TITLE
xuperunion supports a new consensus of XPoS

### LINKED ISSUE #236: XPOS: An Effective and Incentive Mechanism Consensus Based on Proof-of-Stake
**Brief of your idea**
Introduce an effective and incentive mechanism consensus based on Proof-of-Stake encouraging people to participate in the blockchain system.

**What could be better if XuperChain using your idea**
Solve  the problems of the vitality of the sustainable development of the community and the cold start of the network;
A more anti-forking mmechanism with XPower, a weight-based algorithm, adding the cost of malicious attack.

**Are there any side effects?**
None

### PR DESCRIPTION
**Pull request by ToWorld**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #236 

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Brief of your solution

Please describe your solution to solve the issue or feature request.

## How Has This Been Tested?

Regression Test

### CODE REVIEW

File: utxo/utxo.go [hunk 1]
```diff
@@ -623,9 +623,12 @@ func (uv *UtxoVM) SelectUtxos(fromAddr string, fromPubKey string, totalNeed *big
 			}
 			txInputs = append(txInputs, txInput)
 			cacheKeys[uKey] = true
-			if utxoTotal.Cmp(totalNeed) >= 0 {
-				foundEnough = true
-				break
+			// when cancel lease, totalNeed equals negative one
+			if totalNeed != big.NewInt(-1) {
```
Comment (fxsjy) [inline utxo/utxo.go]: is there a better way?

File: cmd/cli/cli.go [hunk 2]
```diff
@@ -490,7 +499,7 @@ func assembleTxInputsSupportAccount(ctx context.Context, client pb.XchainClient,
 		return nil, nil, ErrSelectUtxo
 	}
 	// 多出来的utxo需要再转给自己
-	if utxoTotal.Cmp(totalNeed) > 0 {
+	if totalNeed.Cmp(big.NewInt(-1)) != 0 && utxoTotal.Cmp(totalNeed) > 0 {
```
[the remark this answers is not in the document]
Comment (ToWorld) [reply cmd/cli/cli.go]: Thank you for your advice and I will correct it.

File: cmd/cli/xpos_cancel.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		},
+	}
+
+	c.addFlags()
+	return c.cmd
+}
+
+func (c *XPoSCancelLeaseCommand) addFlags() {
+	c.cmd.Flags().StringVar(&c.from, "from", "", "leased address")
+	c.cmd.Flags().StringVar(&c.to, "to", "", "lease address regularly")
```
[the remark this answers is not in the document]
Comment (ToWorld) [reply cmd/cli/xpos_cancel.go]: Thank you for your advice and I will correct it.

File: utxo/utxo.go (hunk 1 above)
Comment (ToWorld) [reply utxo/utxo.go]: Thank you for your advice and I will correct it.

**Pull request closed without merging**